mesh/mesh.cpp

```diff
@@ -4430,6 +4432,161 @@ void Mesh::MakeSimplicial_(const Mesh &orig_mesh, int *vglobal)
    MFEM_ASSERT(CheckBdrElementOrientation(false) == 0, "");
 }
 
+Mesh Mesh::MakePeriodic(const Mesh &orig_mesh, const std::vector<int> &v2v)
+{
+   Mesh periodic_mesh(orig_mesh, true); // Make a copy of the original mesh
+   const FiniteElementSpace *nodal_fes = orig_mesh.GetNodalFESpace();
+   int nodal_order = nodal_fes ? nodal_fes->GetMaxElementOrder() : 1;
+   periodic_mesh.SetCurvature(nodal_order, true);
+
+   // renumber element vertices
+   for (int i = 0; i < periodic_mesh.GetNE(); i++)
+   {
+      Element *el = periodic_mesh.GetElement(i);
+      int *v = el->GetVertices();
+      int nv = el->GetNVertices();
+      for (int j = 0; j < nv; j++)
+      {
+         v[j] = v2v[v[j]];
+      }
+   }
+   // renumber boundary element vertices
+   for (int i = 0; i < periodic_mesh.GetNBE(); i++)
+   {
+      Element *el = periodic_mesh.GetBdrElement(i);
+      int *v = el->GetVertices();
+      int nv = el->GetNVertices();
+      for (int j = 0; j < nv; j++)
+      {
+         v[j] = v2v[v[j]];
+      }
+   }
+
+   periodic_mesh.RemoveUnusedVertices();
+   return periodic_mesh;
+}
+
+std::vector<int> Mesh::CreatePeriodicVertexMapping(
+   const std::vector<Vector> &translations, double tol) const
+{
+   int sdim = SpaceDimension();
+
+   Vector coord(sdim), at(sdim), dx(sdim);
+   Vector xMax(sdim), xMin(sdim), xDiff(sdim);
+   xMax = xMin = xDiff = 0.0;
+
+   // Get a list of all vertices on the boundary
+   set<int> bdr_v;
+   for (int be = 0; be < GetNBE(); be++)
+   {
+      Array<int> dofs;
+      GetBdrElementVertices(be,dofs);
+
+      for (int i = 0; i < dofs.Size(); i++)
+      {
+         bdr_v.insert(dofs[i]);
+
+         coord = GetVertex(dofs[i]);
+         for (int j = 0; j < sdim; j++)
+         {
+            xMax[j] = max(xMax[j], coord[j]);
+            xMin[j] = min(xMin[j], coord[j]);
+         }
+      }
+   }
+   add(xMax, -1.0, xMin, xDiff);
+   double dia = xDiff.Norml2(); // compute mesh diameter
+
+   // slaves[v] is the index of the master vertex of slaver vertex `v`
```

Comment:
So, I've started to see people move away from this master-slave terminology to describe these sort of periodic conditions. I think it wouldn't hurt for us to consider changing this here. I've seen a number of alternative descriptions such as A/B, first/secondary, or primary/replica. I'm sure there are others as well that could work.

Reply:
I think primary/replica would work well in this context. I can make the changes if people would like.

Reply:
If it is an exact copy, I like `primary/replica`

For more general interpolation, I'll prefer something like `primary/derived` or`true/constrained`.

Reply:
Might want to avoid the `true/constrained` name with `true` being a keyword and all :P

Reply:
I agree @tzanio, I think in this case replica is appropriate because the vertices really are (topologically) exact copies.

Reply:
> Might want to avoid the true/constrained name with true being a keyword and all :P


true that

Reply:
lets go with `primary`/`replica` here and maybe `primary`/`constrained` for the general case

Reply:
I made the changes — @rcarson3 let me know if you think it looks good to you.